src/Domain/Package/PackageManager.php

```diff
@@ -180,29 +195,41 @@ private function initializePackages(FixturePathHandler $fixture_path_handler, st
     foreach ($data as $package_name => $datum) {
       // Skipping a null datum provides for a package to be effectively removed
       // from the active specification at runtime by setting its value to NULL
-      // in the packages configuration alter file.
+      // in the packages' configuration alter file.
       if ($datum === NULL) {
         continue;
       }
 
-      if (is_array($datum) && count($datum) === 0) {
-        $this->addPackage($datum, $fixture_path_handler, $package_name);
+      // Remove packages which have declared core_matrix as something other than
+      // an array even if they declared a valid version as it will cause ORCA to
+      // throw an error in the Options resolver.
+      if ((array_key_exists('core_matrix', $datum) && !is_array($datum['core_matrix']))) {
+        $this->output->writeln('Skipping ' . $package_name . 'as core_matrix not defined as an array.');
         continue;
       }
 
-      if ((array_key_exists('version', $datum) || array_key_exists('version_dev', $datum)) && $this->containsValidVersion($datum)) {
+      // Add packages which have defined an empty array.
+      if (is_array($datum) && count($datum) === 0) {
         $this->addPackage($datum, $fixture_path_handler, $package_name);
         continue;
       }
 
-      if (isset($datum['core_matrix'])) {
+      // Process core_matrix.
+      if (array_key_exists('core_matrix', $datum)) {
         $constraints = array_values($datum['core_matrix']);
         foreach ($constraints as $constraint) {
           if ($this->containsValidVersion($constraint)) {
             $this->addPackage($datum, $fixture_path_handler, $package_name);
             break;
           }
         }
+        continue;
+      }
+
+      // If all above conditions are not met we check to see if either of
+      // version and version_dev are defined.
```

Comment:
At this level of abstraction, all that matters is that the datum contains valid version information--which is already captured in the method name. We don't want to add distracting details here that are sure to get out of sync with the method logic over time.

```suggestion
```